### ISSUE TITLE
Adding a path to the path variable through docker run -e option

### ISSUE THREAD

**Issue by thefragmenter** [moby/moby#35435]
**Description**

TL;DR: How can I add an entry to the already existing path through docker run?

When you try to set the PATH variable through the docker run command like below:

`docker run -it -e PATH=/home/apps:$PATH debian bash`

` Error response from daemon: oci runtime error: container_linux.go:265: starting container process caused "exec: \"bash\": executable file not found in $PATH"`

The value of the $PATH is taken from the host (which seems to be the expected behaviour) however this prevents me from updating the path and forces me to type the entire PATH instead of appending a new entry to the $PATH variable as in only the following works which is not intuitive:

`docker run -it -e PATH=/home/apps:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin debian bash`

I know that the PATH variable can be set using the Dockerfile but that adds an extra step of creating the image. 

**Steps to reproduce the issue:**
1. `docker run -it -e PATH=/home/apps:$PATH debian bash`

**Describe the results you received:**

Error response from daemon: oci runtime error: container_linux.go:265: starting container process caused "exec: \"bash\": executable file not found in $PATH"

**Describe the results you expected:**

Update the PATH variable inside the container instead of using the PATH from host. Is there any escape character that needs to be used?

**Output of `docker version`:**

```
docker version
Client:
 Version:      17.09.0-ce
 API version:  1.32
 Go version:   go1.8.3
 Git commit:   afdb6d4
 Built:        Tue Sep 26 22:40:09 2017
 OS/Arch:      windows/amd64

Server:
 Version:      17.09.0-ce
 API version:  1.32 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   afdb6d4
 Built:        Tue Sep 26 22:45:38 2017
 OS/Arch:      linux/amd64
 Experimental: true
```

**Output of `docker info`:**

```
docker info
Containers: 35
 Running: 1
 Paused: 0
 Stopped: 34
Images: 41
Server Version: 17.09.0-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host ipvlan macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: active
 NodeID: msh93rf4axgjvkwe2mexmyde1
 Is Manager: true
 ClusterID: r9xu9ij5m3s03vnbo1xcjpp1j
 Managers: 1
 Nodes: 1
 Orchestration:
  Task History Retention Limit: 5
 Raft:
  Snapshot Interval: 10000
  Number of Old Snapshots to Retain: 0
  Heartbeat Tick: 1
  Election Tick: 3
 Dispatcher:
  Heartbeat Period: 5 seconds
 CA Configuration:
  Expiry Duration: 3 months
  Force Rotate: 0
 Autolock Managers: false
 Root Rotation In Progress: false
 Node Address: 192.168.65.2
 Manager Addresses:
  192.168.65.2:2377
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 06b9cb35161009dcb7123345749fef02f7cea8e0
runc version: 3f2f8b84a77f73d38244dd690525642a72156c64
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.49-moby
Operating System: Alpine Linux v3.5
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 1.934GiB
Name: moby
ID: 5WML:VEAK:JOMO:XCX7:GHVT:ELUO:6LTJ:MZAV:YLM7:PW4Y:KI5X:BXWF
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 45
 Goroutines: 168
 System Time: 2017-11-08T04:39:47.0653124Z
 EventsListeners: 0
Registry: https://index.docker.io/v1/
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Comment by thaJeztah** [comment]
The docker client does not know about environment variables that are present inside the container, so prepending this at the command-line won't work indeed.

If this is just for an interactive session, something like this should probably work;

```bash
docker run -it debian bash -c 'exec env PATH=/home/app:$PATH bash'
```

Doing the above will evaluate the `PATH=/home/app:$PATH` by the shell inside the container, and after that start a bash session.

Please keep in mind that the GitHub issue tracker is not intended as a general support forum,
but for [reporting bugs and feature requests](https://raw.githubusercontent.com/moby/moby/master/.github/ISSUE_TEMPLATE.md). For other type of questions, consider using one of;

- the Docker Support Forums - https://forums.docker.com
- the [Docker community Slack channel](https://blog.docker.com/2016/11/introducing-docker-community-directory-docker-community-slack/) (register here: [http://dockr.ly/community](http://dockr.ly/community))
- [StackOverflow](https://stackoverflow.com)

I'm closing this issue because this is not a bug, but feel free to continue the conversation
